internal/shim-sgx/src/handler/mod.rs

```diff
@@ -174,4 +178,68 @@ impl<'a> Handler<'a> {
 
         self.ssa.gpr.rip += 2;
     }
+
+    /// Print a stack trace using the SSA registers.
+    fn print_ssa_stack_trace(&mut self) {
+        if DEBUG {
+            unsafe { self.print_stack_trace(self.ssa.gpr.rip, self.ssa.gpr.rbp) }
+        }
+    }
+
+    /// Print out `rip` relative to the shim (S) or the payload (P) base address.
+    ///
+    /// This can be used with `addr2line` and the executable with debug info
+    /// to get the function name and line number.
+    unsafe fn print_rip(&mut self, rip: u64) {
+        let exec_range = &ENARX_EXEC_START as *const u8 as u64..&ENARX_EXEC_END as *const u8 as u64;
```

Comment:
can we perhaps introduce a helper for this cast? There's exact same usage below and it would reduce the cognitive load required to interpret this

Reply:
I don't see any exact usage, but I can offer this:

```diff
diff --git a/internal/shim-sgx/src/handler/mod.rs b/internal/shim-sgx/src/handler/mod.rs
index 19e52e6..8b3967f 100644
--- a/internal/shim-sgx/src/handler/mod.rs
+++ b/internal/shim-sgx/src/handler/mod.rs
@@ -204,6 +204,8 @@ impl<'a> Handler<'a> {
 
     /// Print a stack trace with the old `rbp` stack frame pointers
     unsafe fn print_stack_trace(&mut self, rip: u64, mut rbp: u64) {
+        let shim_range = ENCL_SIZE as u64..&ENARX_HEAP_END as *const _ as u64;
+
         debugln!(self, "TRACE:");
 
         self.print_rip(rip);
@@ -214,13 +216,8 @@ impl<'a> Handler<'a> {
                 break;
             }
 
-            if rbp < ENCL_SIZE as u64 {
-                debugln!(self, "invalid rbp: too low: {:>#016x}", rbp);
-                break;
-            }
-
-            if rbp > &ENARX_HEAP_END as *const _ as u64 {
-                debugln!(self, "invalid rbp: too high: {:>#016x}", rbp);
+            if !shim_range.contains(rbp) {
+                debugln!(self, "invalid rbp: {:>#016x}", rbp);
                 break;
             }
```

Reply:
I was referring to `&ENARX_EXEC_START as *const u8 as u64` part, i.e. the `as *const u8 as u64` specifically, which is performed here twice and once more on L197

Reply:
ah... pushed a fixup